src/main/java/ch/powerunit/extensions/matchers/provideprocessor/helper/ProvidesMatchersWithSameValueHelper.java

```diff
@@ -5,74 +5,94 @@
 
 import javax.lang.model.element.AnnotationMirror;
 import javax.lang.model.element.AnnotationValue;
 import javax.tools.Diagnostic.Kind;
 
+import ch.powerunit.extensions.matchers.common.RessourceLoaderHelper;
+import ch.powerunit.extensions.matchers.provideprocessor.Matchable;
 import ch.powerunit.extensions.matchers.provideprocessor.ProvidesMatchersAnnotatedElementMirror;
 import ch.powerunit.extensions.matchers.provideprocessor.dsl.DSLMethod;
 import ch.powerunit.extensions.matchers.provideprocessor.fields.AbstractFieldDescription;
 
 public final class ProvidesMatchersWithSameValueHelper {
 
+	private static final String HAS_SAME_VALUE_IGNORE_CYCLE = RessourceLoaderHelper
+			.loadRessource(ProvidesMatchersWithSameValueHelper.class, "DSLHasSameValueIgnoreAndCycle.txt");
+
 	private ProvidesMatchersWithSameValueHelper() {
 	}
 
-	private static DSLMethod generateWithSameValueWithParentMatcherIgnore(ProvidesMatchersAnnotatedElementMirror target,
-			boolean hasSuper) {
-		String genericNoParent = target.getSimpleNameOfGeneratedInterfaceMatcherWithGenericNoParent();
-		String simpleNameGenericNoParent = target.getSimpleNameOfGeneratedImplementationMatcherWithGenericNoParent();
-		String simpleNameGenericWithParent = target
-				.getFullyQualifiedNameOfClassAnnotatedWithProvideMatcherWithGeneric();
-		String fullyQualified = target.getFullyQualifiedNameOfGeneratedClass();
-		String withSameValueMethodName = target.getMethodNameDSLWithSameValue();
-		String fullGeneric = target.getFullGeneric();
+	private static DSLMethod generateWithSameValueWithParentMatcherIgnoreAndCycle(
+			ProvidesMatchersAnnotatedElementMirror target, boolean hasSuper) {
 		String argumentForParentBuilder = hasSuper
 				? (target.getParentMirror()
 						.map(p -> p.getWithSameValue(false) + "(other" + (p.supportIgnore() ? ",ignoredFields" : "")
 								+ ")")
 						.orElse("org.hamcrest.Matchers.anything()"))
 				: "";
 		String javadoc = target.generateDefaultJavaDocWithoutDSLStarter(Optional.of(
-				"other the other object to be used as a reference.\n ignoredFields fields name that must be ignored."),
+				"other the other object to be used as a reference.\nprevious the previous object of the call stack of matcher\nignoredFields fields name that must be ignored."),
 				"the DSL matcher", false);
-		List<String> lines = new ArrayList<>();
-		lines.add("java.util.Set<String> ignored = new java.util.HashSet<>(java.util.Arrays.asList(ignoredFields));");
-		lines.add(genericNoParent + " m=new " + simpleNameGenericNoParent + "(" + argumentForParentBuilder + ");");
-		target.getFields().stream().flatMap(ProvidesMatchersWithSameValueHelper::copyField).forEach(lines::add);
-		lines.add("return m;");
-		return of(format("%1$s %2$s.%3$s %4$s", fullGeneric, fullyQualified, genericNoParent, withSameValueMethodName))
-				.withArguments(
-						new String[][] { { simpleNameGenericWithParent, "other" }, { "String...", "ignoredFields" } })
+		List<String> lines = Arrays
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Arrays.asList' due to existing static import 'java.util.Arrays.asList'](https://app.codacy.com/gh/powerunit/powerunit-extensions-matchers/pullRequest?prid=5617368)